src/snowflake/snowpark/dataframe.py

```diff
@@ -1872,15 +1882,42 @@ def join_table_function(
         func_expr = _create_table_function_expression(
             func, *func_arguments, **func_named_arguments
         )
+
+        from snowflake.snowpark import context
+
         if func_expr.aliases:
             join_plan = self._session._analyzer.resolve(
                 TableFunctionJoin(self._plan, func_expr)
             )
             old_cols, new_cols = _get_cols_after_join_table(
                 func_expr, self._plan, join_plan
             )
-            return self._with_plan(Project([*old_cols, *new_cols], join_plan))
-        return DataFrame(self._session, TableFunctionJoin(self._plan, func_expr))
+            names = [*old_cols, *new_cols]
+            if context._use_sql_simplifier:
+                return self._with_plan(
+                    SelectStatement(
+                        from_=SelectTableFunction(
+                            func_expr,
+                            other_plan=self._plan,
+                            analyzer=self._session._analyzer,
+                        ),
+                        analyzer=self._session._analyzer,
+                    ).select(names)
+                )
+            return self._with_plan(Project(names, join_plan))
+
+        if context._use_sql_simplifier:
+            return self._with_plan(
+                SelectStatement(
+                    from_=SelectTableFunction(
+                        func_expr,
+                        other_plan=self._plan,
+                        analyzer=self._session._analyzer,
+                    ),
+                    analyzer=self._session._analyzer,
+                )
+            )
```

Comment:
I'm wondering if the following makes sense:

```
names = None
if func_expr.aliases:
  ...
  names = [*old_cols, *new_cols]

if context._use_sql_simplifier:
  select_plan = self._with_plan(...)
  if names:
    select_plan = select_plan.select(names)
  return self._with_plan(select_plan)

if names:
  return self._with_plan(Project(names, join_plan))
return self._with_plan(TableFunctionJoin(self._plan, func_expr))
```

Main goal is to reduce the duplication of `SelectStatement`